src/sources/statsd/mod.rs

```diff
@@ -147,18 +150,22 @@ pub struct StatsdDeserializer;
 
 impl Deserializer for StatsdDeserializer {
     fn parse(&self, bytes: Bytes) -> crate::Result<SmallVec<[Event; 1]>> {
+        emit!(&BytesReceived {
+            byte_size: bytes.len(),
+            protocol: "udp",
```

Comment:
It seems like this Deserializer is also used for the `TCP` and `Unix` modes ([example](https://cs.github.com/vectordotdev/vector/blob/d19f8bc2087ae324b7636f19ee47a2299e49659f/src/sources/statsd/mod.rs#L228)), so the protocol tag doesn't always fit here. Additionally, the underlying TCP source utility logic already emits `TcpBytesReceived` ([see here](https://cs.github.com/vectordotdev/vector/blob/4629f4ad8c5e9d124f29249835b3c7a0592f4f25/src/sources/util/tcp/mod.rs#L266)), so it looks like this would lead to accidental double emission.

Reply:
👍 I remove it

Reply:
Sorry for the late reply here. I think removing here is correct to fix the double emission for `TCP` mode, but does this leave the `UDP` and `Unix` modes without a `BytesReceived` event? There might be some overlap here with what you decide to do in #11124 (think some underlying logic is shared) to instrument `unix_*` modes.

Reply:
the `UDP` mode has the `BytesReceived` now, and the `Unix` mode as well, only with `datagram`. But shit should probably be discussed in #11124.